pkg/cluster/cluster.go

```diff
@@ -67,3 +71,111 @@ func (ln *localNode) Observe(ckit.Observer) {
 func (ln *localNode) Peers() []peer.Peer {
 	return []peer.Peer{ln.self}
 }
+
+// Clusterer implements the behavior required for operating Flow controllers
+// in a distributed fashion.
+type Clusterer struct {
+	Node Node
+	Mux  *http.ServeMux
+
+	RegisteredComponents map[string]struct{}
+}
+
+// New creates a Clusterer.
+func New(log log.Logger, clusterEnabled bool, addr, joinAddr string) (*Clusterer, error) {
+	// Standalone node.
+	if !clusterEnabled {
+		mux := http.NewServeMux()
+		mux.HandleFunc("/", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			w.Write([]byte("clustering is disabled"))
+			w.WriteHeader(http.StatusBadRequest)
+		}))
+		return &Clusterer{
+			Node: NewLocalNode(addr),
+			Mux:  mux,
+
+			RegisteredComponents: make(map[string]struct{}),
+		}, nil
+	}
+
+	host, portStr, err := net.SplitHostPort(addr)
+	if err != nil {
+		return nil, err
+	}
+	port, err := strconv.Atoi(portStr)
+	if err != nil {
+		return nil, err
+	}
+
+	mux := http.NewServeMux()
+
+	gossipConfig := DefaultGossipConfig
+	gossipConfig.NodeName = uuid.NewString()
```

Comment:
Should this be configurable? Should it default to the hostname? Any other ideas?

Reply:
Yes: configurable but default to the host name. 

The node name should be consistent between process restarts, otherwise you'll pick up an entirely different set of targets when restarting.